File: app/src/main/java/com/ndipatri/solarmonitor/services/solar/dto/Overview.java

package com.ndipatri.solarmonitor.services.solar.dto;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Overview {

    @SerializedName("lastUpdateTime")
    @Expose
    private String lastUpdateTime;
    @SerializedName("lifeTimeData")
    @Expose
    private LifeTimeData lifeTimeData;
    @SerializedName("lastYearData")
    @Expose
    private LastYearData lastYearData;
    @SerializedName("lastMonthData")
    @Expose
    private LastMonthData lastMonthData;
    @SerializedName("lastDayData")
    @Expose
    private LastDayData lastDayData;
    @SerializedName("currentPower")
    @Expose
    private CurrentPower currentPower;
    @SerializedName("measuredBy")
    @Expose
    private String measuredBy;

    public String getLastUpdateTime() {
        return lastUpdateTime;
    }

    public void setLastUpdateTime(String lastUpdateTime) {
        this.lastUpdateTime = lastUpdateTime;
    }

    public LifeTimeData getLifeTimeData() {
        return lifeTimeData;
    }

    public void setLifeTimeData(LifeTimeData lifeTimeData) {
        this.lifeTimeData = lifeTimeData;
    }

    public LastYearData getLastYearData() {
        return lastYearData;
    }

    public void setLastYearData(LastYearData lastYearData) {
        this.lastYearData = lastYearData;
    }

    public LastMonthData getLastMonthData() {
        return lastMonthData;
    }

    public void setLastMonthData(LastMonthData lastMonthData) {
        this.lastMonthData = lastMonthData;
    }

    public LastDayData getLastDayData() {
        return lastDayData;
    }

    public void setLastDayData(LastDayData lastDayData) {
        this.lastDayData = lastDayData;
    }

    public CurrentPower getCurrentPower() {
        return currentPower;
    }

    public void setCurrentPower(CurrentPower currentPower) {
        this.currentPower = currentPower;
    }

    public String getMeasuredBy() {
        return measuredBy;
    }

    public void setMeasuredBy(String measuredBy) {
        this.measuredBy = measuredBy;
    }

}

File: app/src/main/java/com/ndipatri/solarmonitor/providers/solarUpdate/dto/solaredge/Overview.kt
package com.ndipatri.solarmonitor.providers.solarUpdate.dto.solaredge

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Overview {

    @SerializedName("lastUpdateTime")
    @Expose
    var lastUpdateTime: String? = null
    @SerializedName("lifeTimeData")
    @Expose
    var lifeTimeData: LifeTimeData? = null
    @SerializedName("lastYearData")
    @Expose
    var lastYearData: LastYearData? = null
    @SerializedName("lastMonthData")
    @Expose
    var lastMonthData: LastMonthData? = null
    @SerializedName("lastDayData")
    @Expose
    var lastDayData: LastDayData? = null
    @SerializedName("currentPower")
    @Expose
    var currentPower: CurrentPower? = null
    @SerializedName("measuredBy")
    @Expose
    var measuredBy: String? = null

}
